src/ng/filter/orderBy.js

```diff
@@ -6,44 +6,118 @@
- *      3 first characters of a property called `name`). The result of a constant expression
- *      is interpreted as a property name to be used in comparisons (for example `"special name"`
- *      to sort object by the value of their `special name` property). An expression can be
- *      optionally prefixed with `+` or `-` to control ascending or descending sort order
- *      (for example, `+name` or `-name`). If no property is provided, (e.g. `'+'`) then the array
- *      element itself is used to compare where sorting.
- *    - `Array`: An array of function or string predicates. The first predicate in the array
- *      is used for sorting, but when two items are equivalent, the next predicate is used.
+ *    - `Function`: A getter function. Ths function will be called with each item as argument and
+ *      the return value will be used for sorting.
+ *    - `string`: An Angular expression. This expression will be evaluated against each item and the
+ *      result will be used for sorting. For example, use `name` to sort by a property called `name`
+ *      or `name.substring(0, 3)` to sort by the first 3 characters of the `name` property.<br />
+ *      (The result of a constant expression is interpreted as a property name to be used for
+ *      comparison. For example, use `"special name"` to sort by a property called `special name`.)<br />
+ *      An expression can be optionally prefixed with `+` or `-` to control the sorting direction,
+ *      ascending or descending. For example, `+name` or `-name`. If no property is provided, (e.g.
+ *      `'+'` or `'-'`), the collection element itself is used in comparisons.
+ *    - `Array`: An array of function and/or string predicates. If a predicate cannot determine the
+ *      relative order of two items, the next predicate is used as a tie-breaker.
+ *
+ * **Note:** If the predicate is missing or empty then it defaults to `'+'`.
  *
- *    If the predicate is missing or empty then it defaults to `'+'`.
+ * @param {(boolean|Function)=} comparator - The comparator function used to determine the relative
+ *    order of value pairs. If omitted, a built-in comparator will be used. Passing `true` will use
+ *    the built-in comparator, but reverse the order.
  *
- * @param {boolean=} reverse Reverse the order of the array.
- * @returns {Array} Sorted copy of the source array.
+ * @returns {Array} - The sorted array.
  *
  *
  * @example
- * The example below demonstrates a simple ngRepeat, where the data is sorted
- * by age in descending order (predicate is set to `'-age'`).
- * `reverse` is not set, which means it defaults to `false`.
-   <example module="orderByExample">
+ * The example below demonstrates a simple ngRepeat, where the data is sorted by age in descending
+ * order (expression is set to `'-age'`). The `comparator` is not set, which means it defaults to
+ * the built-in comparator.
+ *
+   <example module="orderByExample1">
```

Comment:
Can you name this and the other examples (name="desriptiveName"). That makes them easier to find in the build folder.